esp/smc/SMCLib/TpCommon.cpp

```diff
@@ -119,41 +119,21 @@ extern TPWRAPPER_API bool matchNetAddressRequest(const char* netAddressReg, bool
     return streq(netAddressReg, tpMachine.getConfigNetaddress());
 }
 
-extern TPWRAPPER_API bool validateDropZonePath(const char* dropZoneName, const char* netAddr, const char* pathToCheck)
+extern TPWRAPPER_API bool validateDropZoneHostAndPath(const char* dropZoneName, const char* hostToCheck, const char* pathToCheck)
 {
-    if (isEmptyString(netAddr))
-        throw makeStringException(ECLWATCH_INVALID_INPUT, "NetworkAddress not defined.");
-
+    //Both hostToCheck and pathToCheck should not be empty. For backward compatibility, the dropZoneName may be empty.
+    if (isEmptyString(hostToCheck))
+        throw makeStringException(ECLWATCH_INVALID_INPUT, "Host not defined.");
     if (isEmptyString(pathToCheck))
         throw makeStringException(ECLWATCH_INVALID_INPUT, "Path not defined.");
 
     if (containsRelPaths(pathToCheck)) //Detect a path like: /home/lexis/runtime/var/lib/HPCCSystems/mydropzone/../../../
-        return false;
-
-    bool isIPAddressReq = isIPAddress(netAddr);
-    IArrayOf<IConstTpDropZone> allTpDropZones;
-    CTpWrapper tpWrapper;
-    tpWrapper.getTpDropZones(9999, nullptr, false, allTpDropZones); //version 9999: get the latest information about dropzone
-    ForEachItemIn(i, allTpDropZones)
-    {
-        IConstTpDropZone& dropZone = allTpDropZones.item(i);
-        if (!isEmptyString(dropZoneName) && !streq(dropZoneName, dropZone.getName()))
-            continue;
-
-        StringBuffer directory(dropZone.getPath());
-        addPathSepChar(directory);
+        throw makeStringExceptionV(ECLWATCH_INVALID_INPUT, "Invalid path %s", pathToCheck);
 
-        if (!hasPrefix(pathToCheck, directory, true))
-            continue;
-
-        IArrayOf<IConstTpMachine>& tpMachines = dropZone.getTpMachines();
-        ForEachItemIn(ii, tpMachines)
-        {
-            if (matchNetAddressRequest(netAddr, isIPAddressReq, tpMachines.item(ii)))
-                return true;
-        }
-    }
-    return false;
+    StringBuffer path(pathToCheck);
+    addPathSepChar(path);
+    Owned<IPropertyTree> plane = findDropZonePlaneWithName(dropZoneName, path, hostToCheck, isIPAddress(hostToCheck), false);
```

Comment:
not keen on adding this (backward compatibility) functionality at a lower level.
I think this code should be:
```
    if (isEmptyString(dropZoneName))
    {
        Owned<IPropertyTree> plane = findDropZonePlane(path, hostToCheck, isIPAddress(hostToCheck), false);
        return nullptr != plane;
    }
    else
    {
        Owned<IPropertyTree> plane = getDropZonePlane(dropZoneName);
        if (nullptr == plane)
            return false;
        return validateDropZone(plane, path, hostToCheck, isIPAddress(hostToCheck));
    }
```

With validateDropZone being a new function in dautils.hpp which is code extracted out of the existing findDropZonePlane, so it calls it, and here calls it.